v0/util/cache.js

```diff
@@ -17,12 +17,21 @@ class Cache {
 
     return storeFunction().then(result => {
       // store in cache if the value is valid, else skip
+      if (result && typeof result === "object") {
+        if (result.value && result.age) {
+          return result.value;
+        }
+      }
       if (result) {
         this.cache.set(key, result);
       }
       return result;
```

Comment:
The logic is flawed.

```suggestion
      let retVal = result;
      if (result) {
        if (typeof result === "object" and "value" in result and "age" in result) {
          this.cache.set(key, result.value, result.age);
          retVal = result.value;
        } else {
          this.cache.set(key result);
        }
      }
      return retVal;
```